akka-docs/src/main/paradox/additional/rolling-updates.md

```diff
@@ -0,0 +1,93 @@
+# Rolling Updates
+
+A rolling update is the process of replacing one version of the system with another without downtime.
+The changes can be new code, changed dependencies such as new Akka version, or modified configuration.
+
+In Akka, rolling updates are typically used for a stateful Akka Cluster where you can't run two separate clusters in
+parallel during the update, for example in blue green deployments.
+
+For rolling updates related to Akka dependency version upgrades and the migration guides, please see
+@ref:[Rolling Updates and Akka versions](../project/rolling-update.md)
+
+#### This document covers: 
+* [Serialization Compatibility](#serialization-compatibility)
+* [Cluster Sharding](#cluster-sharding)
+* [Migrating Untyped to Typed](#migrating-untyped-to-typed)
+* [Cluster Shutdown](#cluster-shutdown)
+* [Cluster Configuration Compatibility Check](#cluster-configuration-compatibility-check)
+ 
+## Serialization Compatibility
+
+There are two parts of Akka that need careful consideration when performing an rolling update.
+
+1. Compatibility of remote message protocols. Old nodes may send messages to new nodes and vice versa.
+1. Serialization format of persisted events and snapshots. New nodes must be able to read old data, and
+   during the update old nodes must be able to read data stored by new nodes.
+
+There are many more application specific aspects. It's important to have a strategy for serialization
+format that can be evolved and you can find advice in
+@ref:[Persistence - Schema Evolution](../persistence-schema-evolution.md), which also applies to
```

Comment:
I wonder if this is too strong for some (common) scenario's: for many applications, it might be fine to have some message loss during a rolling upgrade, as long as the application recovers afterwards. Of course that's then likely not strictly '0-impact', but still much better than a 'shut down and restart' upgrade, especially when there's typically no protocol changes.

Does a serialization failure currently just drop the message, or break the whole connection? If the latter, we might want to introduce an option to just drop the message instead (even though that's not without risks of course).

Anyway, just food for thought, not something that needs to go into this PR

Reply:
Good finer point to raise. I agree it's more detailed than may belong here for rolling updates, and this doc isn't about guarantees, more about recommendations and things to be aware of. And from there, finer details can ensue. Should we create a `discuss` issue for it?

Reply:
In https://doc.akka.io/docs/akka/current/serialization.html#serializer-with-string-manifest:

For rolling upgrades if the manifest is unknown/serialization failure:
* If the recommended java.io.NotSerializableException is thrown in `fromBinary` it is treated as a transient problem in the TCP remoting layer, the issue logged and message is dropped 
* Breaks the connection if other exceptions thrown: will tear down the TCP connection because it can be an indication of corrupt bytes from the underlying transport